### ISSUE TITLE
CSV Results don't correspond to graph plot values.

### ISSUE THREAD

**Issue by zutron** [autonomio/talos#280]
- [X] I'm up-to-date with the latest release:
    
      pip install -U talos

- [X] I've confirmed that my Keras model works outside of Talos.

X shape - (94776, 641)
Y shape - (94776,)

p = {'lr': [3.2],
    'first_neuron':[10],
     'shape':['triangle'],
     'hidden_layers':[1],
     'batch_size': [8096],
     'epochs': [25],
     'dropout': [0.1],
     'kernel_initializer': ['normal'],
     'optimizer': [Nadam],
     'losses': [binary_crossentropy],
     'activation':[elu],
     'last_activation': ['sigmoid'],
     'classweights':[{0:1, 1:0.5}]
    }

    def keras_model(x_train, y_train, x_val, y_val, params):
    model = Sequential()
    model.add(Dense(params['first_neuron'], input_dim=x_train.shape[1],
                    activation=params['activation'],
                    kernel_initializer=params['kernel_initializer']))
    
    model.add(Dropout(params['dropout']))

    model.add(Dense(1, activation=params['last_activation'],
                    kernel_initializer=params['kernel_initializer']))
    
    model.compile(loss=params['losses'],
                  optimizer=params['optimizer'](lr=lr_normalizer(params['lr'],params['optimizer'])),
                  metrics=[precision_acc,recall_acc])
    
    history = model.fit(x_train, y_train, 
                        validation_data=[x_val, y_val],
                        batch_size=params['batch_size'],
                        callbacks=[live()],
                        epochs=params['epochs'],
                        verbose=0,
                        class_weight=params['classweights'])

    return history, model

The dataset is an unbalanced binary problem.
-------------------------------------------------------------------------

Here is the csv output line for the above dictionary:

round_epochs | val_loss | val_precision_acc | val_recall_acc | loss | precision_acc | recall_acc | lr | first_neuron | shape | hidden_layers | batch_size | epochs | dropout | kernel_initializer | optimizer | losses | activation | last_activation | classweights
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
25 | 0.623747 | 0.833902 | 0.265479368 | 0.400902 | 0.866431 | 0.277613 | 3.2 | 10 | triangle | 1 | 8096 | 25 | 0.1 | normal | <class 'keras.optimizers.Nadam'> | <function binary_crossentropy at 0x000001B7AC1A6488> | <function elu at 0x000001B7AC1C8EA0> | sigmoid | {0: 1,1: 0.5}

And here are the actual plots generated during the scan:

https://imgur.com/a/UTSutSL

As you can see - the CSV values for precision and recall do not match up to the final values on the graph.  In fact, it looks like maybe the values in the CSV results file are actually the max() Y values that the graph hits (so epoch ~3 for recall, and epoch ~22 for precision).

Obviously the data dictionary is just a single test for this example, but the same holds when doing thousands of iterations.   I have also tried different configurations of the model, but still end up with these types of results.  I'm assuming that the graph is the accurate source, and that the data saved to the csv is not correct.

Or am I doing something wrong?

**Comment by zutron** [comment]
Solved!

I had a look through the code and it turns out "last_epoch_value" parameter must be set to True to log the final value.

From Scan.py
last_epoch_value : bool
        Set to True if the last epoch metric values are logged as opposed
        to the default which is peak epoch values for each round.


I would propose that it should default to True rather than the current False. The result metric values logged in in the output CSV file are irrelevant/misleading under the last_epoch_value= False setting.

**Comment by mikkokotila** [comment]
Thanks. In the latest versions it's `True` by default. Closing here.
